src/index.js

```diff
@@ -11,12 +13,25 @@ const defaultSettings = {
     },
 };
 
+function getEnvironment() {
+    const envFlags = ['-e', '--env'];
+    const args = process.argv;
+    const index = args.findIndex(arg => envFlags.includes(arg)) + 1;
+
+    return (index > 0 && index < args.length) ? process.argv[index] : 'default';
+}
+
 module.exports = class Deviance {
     constructor(settings) {
         this.settings = {};
         Object.entries(defaultSettings).forEach(([k, v]) => {
             this.settings[k] = Object.assign({}, v, settings[k]);
         });
+
+        const env = getEnvironment();
+        const { expectedPath, actualPath } = this.settings.regression;
+        this.settings.regression.expectedPath = path.join(expectedPath, env);
+        this.settings.regression.actualPath = path.join(actualPath, env);
```

Comment:
Initially this was done in the `generateFilenames` method, however doing it here, at a base level allows us to clean up the current environments previous run.